controllers/database_reconciler_helpers.go

```diff
@@ -320,7 +269,7 @@ func (r *DatabaseReconciler) setupConnectivity(ctx context.Context, database *nd
 		Name:      database.Name + "-svc",
 		Namespace: req.Namespace,
 	}
-	targetPort := ndb_api.GetDatabasePortByType(database.Spec.Instance.Type)
+	targetPort := ndb_api.GetDatabasePortByType(database.Status.Type)
```

Comment:
Revisit clone also has a type now

Reply:
Closing as it does not make any difference in provisioning/cloning. Can be revisited later if required.